pkgs/development/python-modules/austin-python/default.nix

```
@@ -0,0 +1,88 @@
+{ stdenv
+, lib
+, buildPythonPackage
+
+, austin ? null  # Avoid circular dependency between austin / austin-python.
+, fetchFromGitHub
+, hatchling
+, hatch-vcs
+, poetry-core
+, protobuf
+, psutil
+, pytestCheckHook
+, pythonOlder
+, pythonRelaxDepsHook
+, toml
+}:
+
+buildPythonPackage rec {
+  pname = "austin-python";
+  version = "1.4.2";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "p403n1x87";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-ddzv17lxj+59NMEe13eUNUmuzezrLY5q08+2Rrocp9E=";
+  };
+
+  postPatch = ''
+    # Fixup paths to dependencies usually found in PATH.
+    substituteInPlace austin/tools/resolve.py \
+      --replace '"addr2line"' '"${stdenv.cc.bintools}/bin/addr2line"'
+    ${if austin == null then ""
+      else ''
+        substituteInPlace austin/__init__.py \
+          --replace 'return self.BINARY' 'return "${austin}/bin/austin"'
+      ''}
+  '';
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    hatchling
+    hatch-vcs
+    poetry-core
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    protobuf
+    psutil
+    toml
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  doCheck = austin != null;
+
+  pythonRelaxDeps = [ "protobuf" ];
+
+  disabledTests = [
+    # With the way we patch austin-python to find austin (from the Nix store
+    # instead of $PATH) the tests always find a proper austin binary, causing
+    # them to fail. This is fine, it just means the error condition this is
+    # trying to test cannot happen with Nix.
```

Please short this comment

---

This might be the first time I'm being told in a code review to have *less* documentation for my code... What's your threshold? 3 lines is ok but 4 isn't?

---

```suggestion
    # the path to the austin binary is hardcoded, so it cannot fail to find it in PATH and the tests always fail
```

The problem is not about the length but the information density.

---

I find this less readable, and this loses nuance that I intended to document:

- The path to the austin binary is hardcoded *by our patching*
- The tests always fail *but it's expected and understood to not be a problem*